### PR TITLE
Week5 - webscraper added

### PR DESCRIPTION
**Pull request by agriaso**
Added basic web scraper using beautiful soup

Pulls data from the target URL, careers.sf.gov

Writes the pulled data to a row in a CSV file

### CODE REVIEW

File: analyze_job_data.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    # Get current date in UTC timezone
+    utc_now = datetime.datetime.now(datetime.UTC)
+
+    # Convert to a specific timezone (for example, 'America/New_York')
+    desired_timezone = pytz.timezone('America/Los_Angeles')
+    today_timezone_aware = utc_now.replace(tzinfo=pytz.utc).astimezone(desired_timezone)
+
+
+    df["Meta_DatePosted"] = pd.to_datetime(df["Meta_DatePosted"])
+    df['Listing_Duration'] = pd.to_datetime(today_timezone_aware) - df['Meta_DatePosted']
```
Comment (heemonsu) [inline analyze_job_data.py:18]: nit: Just a formatting thing, but I would use consistent quotes for string across your code. Just need to pick one and stick to it for all your strings. You can use any auto-formatter like `ruff` or `black` for consistency in code formatting

File: test_analyze_job_data.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from analyze_job_data import analyze_job_data
+import pandas as pd
+
+def test_analyze_job_data():
+    df = pd.read_csv("scraped_data3.csv")
+    analyze_job_data(df)
+    assert True
+
+if __name__ == "__main__":
```
Comment (heemonsu) [inline test_analyze_job_data.py:9]: Why not just run the tests with pytest, do you still find use for the `main` loop here?

File: jobscraper2.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                if itemprop in meta_data:
+                    meta_data[itemprop] = content
+
+            # Process paragraphs and extract job details
+            formatted_p = {}
+            formatted_p.update(split_job_info(joined_paragraphs))
+
+            # Combining all data into a single row
+            row_data = {
+                'Department': formatted_p.get('Department', ''),
```
Comment (heemonsu) [inline jobscraper2.py:143]: This feels weird. Both functions here are basically doing the same thing, but just populating different fields for the row_data. I think you should think about how we can just have a single implementation here.

Maybe you can create a class with method that constructs `formatted_p` once, and allows access with a different function for different use cases? 

We can also think about maybe a general function that takes in a dict like that maps input/output strings`"Meta_Identifier": "identifier"`, but that would still end up doing a qury and going through all divs for each function call, which doesn't seem necessary

**Review by heemonsu — COMMENTED** [review]
Overall seems good. I think the file structure etc. could be a bit better, but its not blocking progress here.

File: jobscraper2.py (hunk 3 above)
Comment (agriaso) [reply jobscraper2.py:143]: I have combined these and used the implementation that doesn't process the paragraphs while scraping the data.

File: test_analyze_job_data.py (hunk 2 above)
Comment (agriaso) [reply test_analyze_job_data.py:9]: I have removed the main loops and implemented proper test formatting for testing the paragraph formatting function, drop duplicates function, and job scraping function.

**Pull request merged**